src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MailParser.java

```diff
@@ -0,0 +1,99 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Describes a parser to extract amazon giftcodes from mails.
+ */
+public class MailParser {
+
+    /**
+     * Extracts a giftcode out of a message.
+     *
+     * @param msg the message, to extract the code
+     * @return returns the giftcode
+     * @throws MoneyTransferException gets thrown, if an error occurred during parsing
+     */
+    protected static GiftCodeRecord parseAmazonGiftCode(Message msg) throws MoneyTransferException {
+        //Extract Message
+        String message;
+        try {
+            if (!msg.getFrom()[0].toString().toLowerCase().contains("amazon.de")) {
```

Comment:
amazon.de@t-online.de is also valid here, maybe you need a better checking here.